pmix2x/libz link failure on NetBSD7 with 3.0.0rc2

@PHHargrove is hitting a pmix2/libz link failure testing 3.0.0rc2 on a NetBSD7 system:

```
I am seeing the following on NetBSD-7.2/amd64:

/bin/sh ../../../libtool  --tag=CC    --mode=link gcc -std=gnu99  -g -finline-functions -fno-strict-aliasing -mcx16 -pthread    -o opal_wrapper opal_wrapper.o ../../../opal/libopen-pal.la -lrt -lexecinfo -lm -lutil
libtool: link: gcc -std=gnu99 -g -finline-functions -fno-strict-aliasing -mcx16 -pthread -o opal_wrapper opal_wrapper.o  ../../../opal/.libs/libopen-pal.a -lpthread -lrt -lexecinfo -lm -lutil -pthread
../../../opal/.libs/libopen-pal.a(compress.o): In function `pmix_util_compress_string':
/home/phargrov/OMPI/openmpi-3.0.0rc2-netbsd7-amd64-static/openmpi-3.0.0rc2/opal/mca/pmix/pmix2x/pmix/src/util/compress.c:41: undefined reference to `deflateInit_'
/home/phargrov/OMPI/openmpi-3.0.0rc2-netbsd7-amd64-static/openmpi-3.0.0rc2/opal/mca/pmix/pmix2x/pmix/src/util/compress.c:44: undefined reference to `deflateBound'
/home/phargrov/OMPI/openmpi-3.0.0rc2-netbsd7-amd64-static/openmpi-3.0.0rc2/opal/mca/pmix/pmix2x/pmix/src/util/compress.c:57: undefined reference to `deflate'
/home/phargrov/OMPI/openmpi-3.0.0rc2-netbsd7-amd64-static/openmpi-3.0.0rc2/opal/mca/pmix/pmix2x/pmix/src/util/compress.c:58: undefined reference to `deflateEnd'
../../../opal/.libs/libopen-pal.a(compress.o): In function `pmix_util_uncompress_string':
/home/phargrov/OMPI/openmpi-3.0.0rc2-netbsd7-amd64-static/openmpi-3.0.0rc2/opal/mca/pmix/pmix2x/pmix/src/util/compress.c:119: undefined reference to `inflateInit_'
/home/phargrov/OMPI/openmpi-3.0.0rc2-netbsd7-amd64-static/openmpi-3.0.0rc2/opal/mca/pmix/pmix2x/pmix/src/util/compress.c:128: undefined reference to `inflate'
/home/phargrov/OMPI/openmpi-3.0.0rc2-netbsd7-amd64-static/openmpi-3.0.0rc2/opal/mca/pmix/pmix2x/pmix/src/util/compress.c:129: undefined reference to `inflateEnd'

It looks like configure found libz:
  checking for library containing deflate... -lz
and yet -lz doesn't appear in the link command.
```
reported on the devel list:
https://www.mail-archive.com/devel@lists.open-mpi.org//msg20325.html

--- a/config/opal_setup_zlib.m4
+++ b/config/opal_setup_zlib.m4
@@ -56,7 +56,7 @@ AC_DEFUN([OPAL_ZLIB_CONFIG],[
                            [$opal_zlib_libdir],
                            [opal_zlib_support=1],
                            [opal_zlib_support=0])
-        if test $opal_zlib_support == "1"; then
+        if test $opal_zlib_support = "1"; then
             LIBS="$LIBS -lz"
             if test "$opal_zlib_standard_header_location" != "yes"; then
                 CPPFLAGS="$CPPFLAGS $opal_zlib_CPPFLAGS"

--- a/opal/mca/pmix/pmix2x/configure.m4
+++ b/opal/mca/pmix/pmix2x/configure.m4
@@ -53,7 +53,7 @@ AC_DEFUN([MCA_opal_pmix_pmix2x_CONFIG],[
                   [AC_HELP_STRING([--enable-pmix-timing],
                                   [Enable PMIx timing measurements (default: disabled)])])
     AC_MSG_CHECKING([if PMIx timing is enabled])
-    if test "$enable_pmix_timing" == "yes"; then
+    if test "$enable_pmix_timing" = "yes"; then
         AC_MSG_RESULT([yes])
         opal_pmix_pmix2x_timing_flag=--enable-pmix-timing
     else

--- a/opal/mca/pmix/pmix2x/pmix/config/pmix.m4
+++ b/opal/mca/pmix/pmix2x/pmix/config/pmix.m4
@@ -933,7 +933,7 @@ AC_MSG_CHECKING([if want shared memory datastore])
 AC_ARG_ENABLE([dstore],
               [AC_HELP_STRING([--disable-dstore],
                               [Using shared memory datastore (default: enabled)])])
-if test "$enable_dstore" == "no" ; then
+if test "$enable_dstore" = "no" ; then
     AC_MSG_RESULT([no])
     WANT_DSTORE=0
 else
@@ -952,7 +952,7 @@ AC_MSG_CHECKING([if want dstore pthread-based locking])
 AC_ARG_ENABLE([dstore-pthlck],
               [AC_HELP_STRING([--disable-dstore-pthlck],
                               [Disable pthread-based lockig in dstor (default: enabled)])])
-if test "$enable_dstore_pthlck" == "no" ; then
+if test "$enable_dstore_pthlck" = "no" ; then
     AC_MSG_RESULT([no])
     DSTORE_PTHREAD_LOCK="0"
 else

--- a/opal/mca/pmix/pmix2x/pmix/config/pmix_check_lock.m4
+++ b/opal/mca/pmix/pmix2x/pmix/config/pmix_check_lock.m4
@@ -5,6 +5,7 @@ dnl                         All rights reserved.
 dnl Copyright (c) 2017      IBM Corporation.  All rights reserved.
 dnl Copyright (c) 2017      Research Organization for Information Science
 dnl                         and Technology (RIST). All rights reserved.
+dnl Copyright (c) 2017      Intel, Inc. All rights reserved.
 dnl $COPYRIGHT$
 dnl
 dnl Additional copyrights may follow
@@ -26,7 +27,7 @@ AC_DEFUN([PMIX_CHECK_DSTOR_LOCK],[
         _x_ac_fcntl_lock_found="1"
     ], [], [#include <fcntl.h>])
 
-    if test "$DSTORE_PTHREAD_LOCK" == "1"; then
+    if test "$DSTORE_PTHREAD_LOCK" = "1"; then
         AC_CHECK_FUNC([pthread_rwlockattr_setkind_np],
             [AC_EGREP_HEADER([PTHREAD_RWLOCK_PREFER_WRITER_NONRECURSIVE_NP],
                     [pthread.h],[
@@ -43,15 +44,15 @@ AC_DEFUN([PMIX_CHECK_DSTOR_LOCK],[
             ])
         ])
 
-        if test "$_x_ac_pthread_lock_found" == "0"; then
-            if test "$_x_ac_fcntl_lock_found" == "1"; then
+        if test "$_x_ac_pthread_lock_found" = "0"; then
+            if test "$_x_ac_fcntl_lock_found" = "1"; then
                 AC_MSG_WARN([dstore: pthread-based locking not found, will use fcntl-based locking.])
             else
                 AC_MSG_ERROR([dstore: no available locking mechanisms was found. Can not continue. Try disabling dstore])
             fi
         fi
     else
-        if test "$_x_ac_fcntl_lock_found" == "0"; then
+        if test "$_x_ac_fcntl_lock_found" = "0"; then
             AC_MSG_ERROR([dstore: no available locking mechanisms was found. Can not continue. Try disabling dstore])
         fi
         LIBS="$orig_libs"

--- a/opal/mca/pmix/pmix2x/pmix/config/pmix_setup_zlib.m4
+++ b/opal/mca/pmix/pmix2x/pmix/config/pmix_setup_zlib.m4
@@ -56,7 +56,7 @@ AC_DEFUN([PMIX_ZLIB_CONFIG],[
                            [$pmix_zlib_libdir],
                            [pmix_zlib_support=1],
                            [pmix_zlib_support=0])
-        if test $pmix_zlib_support == "1"; then
+        if test $pmix_zlib_support = "1"; then
             LIBS="$LIBS -lz"
             PMIX_EMBEDDED_LIBS="$PMIX_EMBEDDED_LIBS -lz"
             if test "$pmix_zlib_standard_header_location" != "yes"; then

